ui/pages/confirm-add-suggested-nft/confirm-add-suggested-nft.js

```diff
@@ -0,0 +1,296 @@
+} from '../../../shared/constants/metametrics';
+import { AssetType } from '../../../shared/constants/transaction';
+import {
+  BUTTON_SIZES,
+  ButtonIcon,
+  ButtonIconSize,
+  ButtonLink,
+  IconName,
+  Text,
+} from '../../components/component-library';
+import {
+  getCurrentChainId,
+  getRpcPrefsForCurrentProvider,
+  getSuggestedNfts,
+  getIpfsGateway,
+} from '../../selectors';
+import NftDefaultImage from '../../components/app/nft-default-image/nft-default-image';
+import { getAssetImageURL, shortenAddress } from '../../helpers/utils/util';
+import {
+  BorderRadius,
+  IconColor,
+  TextAlign,
+  TextVariant,
+} from '../../helpers/constants/design-system';
+import Box from '../../components/ui/box/box';
+
+const ConfirmAddSuggestedNFT = () => {
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const suggestedNfts = useSelector(getSuggestedNfts);
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+  const chainId = useSelector(getCurrentChainId);
+  const ipfsGateway = useSelector(getIpfsGateway);
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const handleAddNftsClick = useCallback(async () => {
+    await Promise.all(
+      suggestedNfts.map(async ({ requestData: { asset }, id }) => {
+        await dispatch(resolvePendingApproval(id, null));
+
+        trackEvent({
+          event: MetaMetricsEventName.NftAdded,
+          category: MetaMetricsEventCategory.Wallet,
+          sensitiveProperties: {
+            token_symbol: asset.symbol,
+            token_id: asset.tokenId,
+            token_contract_address: asset.address,
+            source_connection_method: MetaMetricsTokenEventSource.Dapp,
+            token_standard: asset.standard,
+            asset_type: AssetType.NFT,
+          },
+        });
+      }),
+    );
+    history.push(mostRecentOverviewPage);
+  }, [dispatch, history, trackEvent, mostRecentOverviewPage, suggestedNfts]);
+
+  const handleCancelNftClick = useCallback(async () => {
+    await Promise.all(
+      suggestedNfts.map(async ({ id }) => {
+        return dispatch(
+          rejectPendingApproval(
+            id,
+            serializeError(ethErrors.provider.userRejectedRequest()),
+          ),
+        );
+      }),
+    );
+    history.push(mostRecentOverviewPage);
+  }, [dispatch, history, mostRecentOverviewPage, suggestedNfts]);
+
+  const goBackIfNoSuggestedNftsOnFirstRender = () => {
+    if (!suggestedNfts.length) {
+      history.push(mostRecentOverviewPage);
+    }
+  };
+
+  useEffect(() => {
+    goBackIfNoSuggestedNftsOnFirstRender();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  let origin;
+  if (suggestedNfts.length) {
+    try {
+      origin = new URL(suggestedNfts[0].origin)?.host;
+    } catch {
+      origin = 'dapp';
+    }
+  }
+  return (
+    <div className="page-container">
```

Comment:
Can we get these divs replaced with our Box component to utilize our design system and not add debt? I know you are copying over from other confirmations but would help us keep moving in the right direction. @jpuri do you want to review these new confirmation screens?

Reply:
done [here](https://github.com/MetaMask/metamask-extension/pull/19454/commits/19234ca62e07ac3b5a83f7181f4f946e68100b43)